add YAML file to .github/workflows/ to use Github Actions for CI

gotanygrapes-lemonade opened the pull request.

Commit:
add YAML file to .github/workflows/ to use Github Actions for CI

orien commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,33 @@
+---
+name: tests
```
I think we should name this file `tests.yml` to match the defined name.

gotanygrapes-lemonade replied:
Thank you @orien, I renamed `test.yml` to `test.yml`. Should I change the file name to `test.yml` in the badge code in the README? 

`[![tests](https://github.com/envato/github-pull-request-buildkite-plugin/actions/workflows/test.yml/badge.svg)](https://github.com/envato/github-pull-request-buildkite-plugin/actions/workflows/test.yml)`

orien replied:
No, tests is good 👍

orien commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,33 @@
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
```
Rather than pointing at the master branch (which could change at any time),  I think we should point to a tag so we can control when things change.

There are a few tags here: https://github.com/ludeeus/action-shellcheck/tags
The latest is probably best.

It also seems this line is missing a new-line character. Try adding a line.

orien replied:
It seems this step is failing the build. The [readme suggests](https://github.com/ludeeus/action-shellcheck#run-shellcheck-with-all-paths-in-a-single-invocation) using this option:

```yaml
    - name: shellcheck
      uses: ludeeus/action-shellcheck@1.1.0
      with:
        check_together: 'yes'
```

orien reviewed (commented):
Awesome! Thanks Bing.

I notice a badge in the project readme showing the status of the Buildkite build. Let's update that to use a badge from GitHub Actions.

https://user-images.githubusercontent.com/497874/151882665-0bb258ec-a308-4086-9fce-8e5e08d0dafe.mov

Commit:
Apply changes Orien suggested:

- change the badge in README
- change the file name from test.yml to tests.yml
- point to tag 1.1.0
- add an new line at the end of the file

orien commented on README.md:
```diff
@@ -1,7 +1,6 @@
 
-![Build status](https://badge.buildkite.com/03e8876c9dbec4eb293aba87dc657e2dadf6afc6c0b9b761b4.svg)
-[![MIT License](https://img.shields.io/badge/License-MIT-brightgreen.svg)](LICENSE)
+[![tests](https://github.com/envato/github-pull-request-buildkite-plugin/actions/workflows/tests.yml/badge.svg)](https://github.com/envato/github-pull-request-buildkite-plugin/actions/workflows/tests.yml)
```
One slight improvement. This ensures the badge only shows the status of the `main` branch.

```suggestion
[![tests](https://github.com/envato/github-pull-request-buildkite-plugin/actions/workflows/tests.yml/badge.svg?branch=main)](https://github.com/envato/github-pull-request-buildkite-plugin/actions/workflows/tests.yml)
```

orien reviewed (approved):
Nice, thanks.

gotanygrapes-lemonade committed:
Update README.md

Co-authored-by: Orien Madgwick <_@orien.io>
